Tasks/Common/webdeployment-common-v2/fileTransformationsUtility.ts

```diff
@@ -54,54 +54,58 @@ export function advancedFileTransformations(isFolderBasedDeployment: boolean, ta
             throw Error(tl.loc("CannotPerformXdtTransformationOnNonWindowsPlatform"));
         }
         else {
-            let isTransformationApplied: boolean = true;
-            if(transformationRules.length > 0) {                
-                transformationRules.forEach(function(rule) {
-                    var args = ParameterParser.parse(rule);
-                    if(Object.keys(args).length < 2 || !args["transform"] || !args["xml"]) {
-                       tl.error(tl.loc("MissingArgumentsforXMLTransformation"));
-                    }
-                    else if(Object.keys(args).length > 2) {
-                        isTransformationApplied = xdtTransformationUtility.specialXdtTransformation(folderPath, args["transform"].value, args["xml"].value, args["result"].value) && isTransformationApplied;
-                    }
-                    else {
-                        isTransformationApplied = xdtTransformationUtility.specialXdtTransformation(folderPath, args["transform"].value, args["xml"].value) && isTransformationApplied;
-                    }
-                });
-            }
-            else{   
-                var environmentName = tl.getVariable('Release.EnvironmentName');             
-                let transformConfigs = ["Release.config"];
-                if(environmentName && environmentName.toLowerCase() != 'release') {
-                    transformConfigs.push(environmentName + ".config");
+            let isTransformationApplied: boolean = true;              
+            transformationRules.forEach(function(rule) {
+                var args = ParameterParser.parse(rule);
+                if(Object.keys(args).length < 2 || !args["transform"] || !args["xml"]) {
+                    tl.error(tl.loc("MissingArgumentsforXMLTransformation"));
                 }
-                isTransformationApplied = xdtTransformationUtility.basicXdtTransformation(folderPath, transformConfigs);
-            }
+                else if(Object.keys(args).length > 2) {
+                    isTransformationApplied = xdtTransformationUtility.specialXdtTransformation(folderPath, args["transform"].value, args["xml"].value, args["result"].value) && isTransformationApplied;
+                }
+                else {
+                    isTransformationApplied = xdtTransformationUtility.specialXdtTransformation(folderPath, args["transform"].value, args["xml"].value) && isTransformationApplied;
+                }
+            });
             
             if(isTransformationApplied) {
                 console.log(tl.loc("XDTTransformationsappliedsuccessfully"));
+            }
+            else {
+                tl.error(tl.loc('FailedToApplySpecialTransformation'));
```

Comment:
There can be two causes of failures 
- Transformation step failed with some error.
- Transformation is already applied 

Error message should reflect clear differentiation between two.